app/util/Utils.scala

```diff
@@ -0,0 +1,19 @@
+package util
+
+import io.circe.ParsingFailure
+import models.{InvalidJsonError, NoRequestBodyError, ProductionMetricsError, UnexpectedExceptionError}
+
+object Utils {
+
+  def processException(exception: Exception): Either[ProductionMetricsError, Nothing] = {
+    val error = exception match {
+      case e: ParsingFailure => InvalidJsonError(e.message)
+      case _ => UnexpectedExceptionError
```

Comment:
It looks like you'll be hiding the exception here and instead reporting something unexpected happened. I'm not sure how useful that would be, you'll have the message in the `Exception` so you could log it out.